src/commands/classified/external.rs

```diff
@@ -67,13 +67,13 @@ pub(crate) async fn run_external_command(
 async fn run_with_iterator_arg(
     command: ExternalCommand,
     context: &mut Context,
-    input: InputStream,
+    input: Option<InputStream>,
     is_last: bool,
-) -> Result<InputStream, ShellError> {
+) -> Result<Option<InputStream>, ShellError> {
     let name = command.name;
     let args = command.args;
     let name_tag = command.name_tag;
-    let inputs = input.into_vec().await;
+    let inputs = input.unwrap_or_else(InputStream::empty).into_vec().await;
```

Comment:
Should we return a more informative error? This will result in:

```
error: External $it needs string data
- shell:1:7
1 | cat $it
  |     ^^^ given row instead of string data
```

Reply:
Yes, ultimately I think we will. The need for an input stream should be part of the Signature, so its presence would be checked by the type checker.

Just to get us going, I think we're probably fine doing this.

Reply:
For that error message, we might want to add hints to our error system, so we can say "hint: you may want to access a part of the row rather than the whole thing eg) cat $it.item"